autometa/validation/datasets.py

```diff
@@ -0,0 +1,174 @@
+    """
+    Parameters
+    ----------
+    folder : str
+        specifies the folder that the user would like to download from, if any
+    file : str
+        specifies the file that the user would like to download, if any
+
+    Returns
+    -------
+    dict
+        nested dictionary with the user's specified files' folder, file, and file_id
+        format: {'folder': {<index>: <folder>}, 'file': {<index>: <file>}, 'file_id': {<index>: <file_id>}}
+
+    """
+    # create the dataframe here
+    index = pd.read_csv("gdown_fileIDs.csv", dtype=str)
+
+    # retrieve only the entries that the user wants
+    if folder == None:
+        target_folder = index
+    else:
+        target_folder = index.query(f'folder == "{folder}"')
+
+    if file == None:
+        target_files = target_folder
+    else:
+        target_files = target_folder.query(f'file == "{file}"')
+
+    targets_dict = target_files.to_dict()
+    return targets_dict
+
+
+def download(targets_dict, out_dirpath: str) -> None:
+
+    """Downloads the files specified in a dictionary.
+
+    Parameters
+    ----------
+    targets_dict : dict
+        nested dictionary with the user's specified files' folder, file, and file_id
+        format: {'folder': {<index>: <folder>}, 'file': {<index>: <file>}, 'file_id': {<index>: <file_id>}}
+    out_dirpath : str
+        directory path where the user wants to download the file(s)
+
+    Returns
+    -------
+    None
+        download is completed through gdown
+
+    """
+
+    key_list = [*targets_dict["file"]]
+    for key in key_list:
+        # retrieve values from targets_dict
+        file = targets_dict["file"][key]
+        folder = targets_dict["folder"][key]
+        file_id = targets_dict["file_id"][key]
+
+        # construct file id into a url to put into gdown
+        url = f"https://drive.google.com/uc?id={file_id}"
+        filename = f"{folder}_{file}"
+        out_fpath = os.path.join(
+            out_dirpath, filename
+        )  # this returns an error when relative file paths get used, eg "~/download"
+
+        # download the specified file with gdown
+        gdown.download(url, out_fpath)
+
+
+def main():
+    import argparse
+    import logging as logger
+
+    logger.basicConfig(
+        format="[%(asctime)s %(levelname)s] %(name)s: %(message)s",
+        datefmt="%m/%d/%Y %I:%M:%S %p",
+        level=logger.DEBUG,
+    )
+
+    parser = argparse.ArgumentParser(
+        prog="autometa-download-dataset",
+        description="Download a simulated community file from google drive to a specified directory",
+    )
+    parser.add_argument(
+        "--folder",
+        help="specify a simulated community size to download from (leave blank to download all)",
+        choices=[
+            "78.125Mbp",
+            "156.25Mbp",
+            "312.5Mbp",
+            "625Mbp",
+            "1250Mbp",
+            "2500Mbp",
+            "5000Mbp",
+            "10000Mbp",
+        ],
+    )
```

Comment:
This might be nitpicky, but does `dataset` seem more appropriate?

`autometa-download --dataset 78.125Mbp`

v.s.

`autometa-download --folder 78.125Mbp`

Also, I think this should be a required argument where the user must explicitly specify a dataset. Perhaps another choice here could be `"all"` where the user has to explicitly state that they would like to download all of the datasets. This way you could add the extra keyword argument to the `add_argument(...)` method

```suggestion
        "--dataset",
        help="specify a simulated community size to download",
        choices=[
            "78.125Mbp",
            "156.25Mbp",
            "312.5Mbp",
            "625Mbp",
            "1250Mbp",
            "2500Mbp",
            "5000Mbp",
            "10000Mbp",
            "all"
        ],
        required=True,
    )
```